tests/test_polynomial.py

```diff
@@ -5,141 +5,90 @@
-        p = Polynomial(1, 2, 3)
-        self.assertEqual((p + 5).coeffs, [6, 2, 3])
-
-    def test_call(self):
-        # Test evaluation of polynomial
-        p = Polynomial(1, 2, 3)  # 1 + 2X + 3X^2
-        self.assertEqual(p(0), 1)
-        self.assertEqual(p(1), 6)
-        self.assertEqual(p(2), 17)
+        # Test addition of polynomials and a polynomial with a scalar
+        self.assertEqual((self.p1 + self.p2).coeffs, [4, 4, 4])
+        self.assertEqual((self.p1 + self.scalar).coeffs, [6, 2, 3])
 
-    def test_eq(self):
-        # Test equality of two polynomials
-        p1 = Polynomial(1, 2, 3)
-        p2 = Polynomial(1, 2, 3)
-        p3 = Polynomial(3, 2, 1)
-        self.assertTrue(p1 == p2)
-        self.assertFalse(p1 == p3)
-
-    def test_eq_scalar(self):
-        # Test equality of polynomial and scalar
-        p = Polynomial(5)
-        self.assertTrue(p == 5)
-        self.assertFalse(p == 4)
-
-    def test_mul(self):
-        # Test multiplication of two polynomials
-        p1 = Polynomial(1, 2)
-        p2 = Polynomial(3, 4)
-        self.assertEqual((p1 * p2).coeffs, [3, 10, 8])
-
-    def test_mul_scalar(self):
-        # Test multiplication of polynomial and scalar
-        p = Polynomial(1, -2, 3)
-        self.assertEqual((p * 3).coeffs, [3, -6, 9])
+    def test_sub(self):
+        # Test subtraction of polynomials and a polynomial with a scalar
+        self.assertEqual((self.p1 - self.p2).coeffs, [-2, 0, 2])
+        self.assertEqual((self.p1 - self.scalar).coeffs, [-4, 2, 3])
 
     def test_neg(self):
-        # Test negation of polynomial
-        p = Polynomial(1, -2, 3)
-        self.assertEqual((-p).coeffs, [-1, 2, -3])
+        # Test negation of a polynomial
+        self.assertEqual((-self.p1).coeffs, [-1, -2, -3])
 
-    def test_pow(self):
-        # Test exponentiation of polynomial
-        p1 = Polynomial(1, 2)
-        with self.assertRaises(NotImplementedError):
-            p2 = p1 ** 2
+    def test_mul(self):
+        # Test multiplication of polynomials and a polynomial with a scalar
+        self.assertEqual((self.p1 * self.p2).coeffs, [3, 8, 14, 8, 3])
+        self.assertEqual((self.p1 * self.scalar).coeffs, [5, 10, 15])
 
     def test_radd(self):
-        # Test right-hand addition
-        p1 = Polynomial(1, 2, 3)
-        p2 = 5 + p1
-        self.assertEqual(p2.coeffs, [6, 2, 3])
-
-    def test_repr(self):
-        # Test representation
-        p = Polynomial(1, -2, 3)
-        self.assertEqual(repr(p), "Polynomial([1, -2, 3])")
+        # Test right-hand addition (scalar + polynomial)
+        self.assertEqual((self.scalar + self.p1).coeffs, [6, 2, 3])
 
     def test_rmul(self):
-        # Test right-hand multiplication
-        p1 = Polynomial(1, 2, 3)
-        p2 = 5 * p1
-        self.assertEqual(p2.coeffs, [5, 10, 15])
+        # Test right-hand multiplication (scalar * polynomial)
+        self.assertEqual((self.scalar * self.p1).coeffs, [5, 10, 15])
```

Comment:
This became significantly less readable. You don't need a whole set up do define an int variable. They are usually done for larger objects. I don't think there's any reason to reuse an int variable.